sdk/core/switcher/entry.S

```diff
@@ -126,6 +133,46 @@ switcher_scheduler_entry_csp:
  * and integer register.  All three registers are clobbered.
  */
 .macro zero_stack base top scratch
+#ifdef CHERIOT_HAS_ZTOP
+	// Derive the capability to the range that should be zeroed.
+	// This is stored in base, leaving top as a second scratch register to use
+	csub               \top, c\top, c\base
+	csetboundsexact    c\base, c\base, \top
```

Comment:
As per discussions out of band, this should probably be
```suggestion
	csetbounds    c\base, c\base, \top
	cincoffset    c\base, c\base, \top
```

- We need to set the address to the top of the region to be zeroed, since the state machine counts down.
- Using inexact bounds allows us to tolerate large (> 2KiB) stacks without risk, and...
  - Because the stack is guaranteed to be a superset representable region, we'll never try to grow beyond it
  - Because the state machine counts down from the (precise) address, it won't zeroize parts of the stack in use
  - It might zero a little bit past the HWM, but that's fine.